web/vars_native.go

```diff
@@ -0,0 +1,45 @@
+// +build go1.7
+
+package web
+
+import (
+	"context"
+	"net/http"
+)
+
+// Vars is a helper function for accessing route
+// parameters from any server.Router implementation. This is the equivalent
+// of using `mux.Vars(r)` with the Gorilla mux.Router.
+func Vars(r *http.Request) map[string]string {
+	// vars doesnt exist yet, return empty map
+	rawVars := r.Context().Value(varsKey)
+	if rawVars == nil {
+		return map[string]string{}
+	}
+
+	// for some reason, vars is wrong type, return empty map
+	vars, ok := rawVars.(map[string]string)
+	if !ok {
+		return map[string]string{}
```

Comment:
I noticed the gorilla/existing one doesn't do any checks on the way out after I pushed this.

Since returning `nil` or an invalid type here will cause a panic (see: https://play.golang.org/p/CV0qEiuDws), I'll go ahead and add a similar check.